proto/regen/ecocredit/v1alpha1/tx.proto

```diff
@@ -52,6 +53,30 @@ service Msg {
 
   // AllowAskDenom is a governance operation which authorizes a new ask denom to be used in sell orders
   rpc AllowAskDenom (MsgAllowAskDenom) returns (MsgAllowAskDenomResponse);
+
+  // CreateBasket creates a bank denom which wraps credits.
+  rpc CreateBasket(MsgCreateBasket) returns (MsgCreateBasketResponse);
+
+  // AddToBasket adds credits to a basket in return for basket tokens.
+  rpc AddToBasket(MsgAddToBasket) returns (MsgAddToBasketResponse);
+
+  // TakeFromBasket takes credits from a basket without regard for which
+  // credits they are. The credits will be auto-retired if retire_on_take
+  // is true. Credits will be chosen randomly using the previous block hash
+  // as a consensus source of randomness.
```

Comment:
@aaronc in terms of implementation, would the following work:
- take the previous block hash and convert it to uint64
- given the total number of different credits within the basket `n`, the first credits that will get picked correspond to: hash modulo n (in terms of order)
- then if we need to take more credits, we get some from the next one and so on

Reply:
That makes sense 👍 . Feel free to suggest an update to the docs to specifiy this more clearly